contractcourt/htlc_success_resolver.go

```diff
@@ -201,50 +238,168 @@ func (h *htlcSuccessResolver) Resolve() (ContractResolver, error) {
+	// If we have non-nil SignDetails, this means that have a 2nd level
+	// HTLC transaction that is signed using sighash SINGLE|ANYONECANPAY
+	// (the case for anchor type channels). In this case we can re-sign it
+	// and attach fees at will. We let the sweeper handle this job.
+	case h.htlcResolution.SignDetails != nil:
+		log.Infof("%T(%x): offering second-layer transition tx to sweeper: %v",
+			h, h.htlc.RHash[:], spew.Sdump(h.htlcResolution.SignedSuccessTx))
+
+		secondLevelInput := input.MakeHtlcSecondLevelSuccessAnchorInput(
+			h.htlcResolution.SignedSuccessTx, h.htlcResolution.SignDetails,
+			h.htlcResolution.Preimage, h.broadcastHeight,
+		)
+		_, err := h.Sweeper.SweepInput(
+			&secondLevelInput,
+			sweep.Params{
+				Fee: sweep.FeePreference{
+					ConfTarget: secondLevelConfTarget,
+				},
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
 
-	// We'll now broadcast the second layer transaction so we can kick off
-	// the claiming process.
-	//
-	// TODO(roasbeef): after changing sighashes send to tx bundler
-	label := labels.MakeLabel(
-		labels.LabelTypeChannelClose, &h.ShortChanID,
-	)
-	err := h.PublishTx(h.htlcResolution.SignedSuccessTx, label)
-	if err != nil {
-		return nil, err
-	}
+		// Wait for the second level transaction to confirm.
+		// TODO(halseth): could use sweeper result?
+		spendNtfn, err := h.Notifier.RegisterSpendNtfn(
+			&h.htlcResolution.SignedSuccessTx.TxIn[0].PreviousOutPoint,
+			h.htlcResolution.SignDetails.SignDesc.Output.PkScript,
+			h.broadcastHeight,
+		)
+		if err != nil {
+			return nil, err
+		}
 
-	// Otherwise, this is an output on our commitment transaction. In this
-	// case, we'll send it to the incubator, but only if we haven't already
-	// done so.
-	if !h.outputIncubating {
-		log.Infof("%T(%x): incubating incoming htlc output",
+		log.Infof("%T(%x): waiting for second-level HTLC to confirm",
 			h, h.htlc.RHash[:])
 
-		err := h.IncubateOutputs(
-			h.ChanPoint, nil, &h.htlcResolution,
-			h.broadcastHeight,
+		var (
+			spend *chainntnfs.SpendDetail
+			ok    bool
+		)
+		select {
+		case spend, ok = <-spendNtfn.Spend:
+			if !ok {
+				return nil, errResolverShuttingDown
+			}
+
+		case <-h.quit:
+			return nil, errResolverShuttingDown
+		}
+
+		// Now that the second level transaction has confirmed, we
+		// report the resolver has moved the next stage.
+		h.reportLock.Lock()
+		h.currentReport.Stage = 2
+		h.reportLock.Unlock()
+
+		waitHeight := uint32(spend.SpendingHeight) +
+			h.htlcResolution.CsvDelay - 1
+		log.Infof("%T(%x): waiting for CSV lock to expire at height %v",
+			h, h.htlc.RHash[:], waitHeight)
+
+		if err := waitForHeight(waitHeight, h.Notifier, h.quit); err != nil {
+			return nil, err
+		}
+
+		// Find output index of the second-level, as the sweeper might
+		// have re-arranged outputs.
+		txOut := h.htlcResolution.SignedSuccessTx.TxOut[0]
+		index := -1
+		for i, o := range spend.SpendingTx.TxOut {
```

Comment:
IIRC, there's a method for this in the `input` pacakge.

Reply:
Nice, done!